src/nvim/syntax.c

```diff
@@ -44,6 +64,10 @@
 #include "nvim/os/time.h"
 
 static bool did_syntax_onoff = false;
+static int invalid_group_id = -1;
+
+// TODO(teto): use array instead GA_EMPTY_INIT_VALUE ?
+static garray_T changed_highlights  = { 0, 0, sizeof(int), 10, NULL };
```

Comment:
I converted the garray into an Array to remove the conversion needed when sending the event.